synfig-studio/src/gui/duckmatic.cpp

```diff
@@ -810,38 +810,33 @@ Duckmatic::set_guides_color(const synfig::Color &c)
 		signal_grid_changed().emit();
 	}
 }
-
-Duckmatic::GuideList::iterator
-Duckmatic::find_guide_x(synfig::Point pos, float radius)
+double
+Duckmatic::calculate_distance_from_guide(const Guide guide, const synfig::Point point)
 {
-	GuideList::iterator iter,best(guide_list_x_.end());
-	float dist(radius);
-	for(iter=guide_list_x_.begin();iter!=guide_list_x_.end();++iter)
-	{
-		float amount(std::fabs(*iter-pos[0]));
-		if(amount<dist)
-		{
-			dist=amount;
-			best=iter;
-		}
-	}
-	return best;
+	return std::fabs((cos(guide.angle.get())*(guide.point[1]-point[1])) - (sin(guide.angle.get())*(guide.point[0] - point[0])));
 }
 
+
 Duckmatic::GuideList::iterator
-Duckmatic::find_guide_y(synfig::Point pos, float radius)
+Duckmatic::find_guide(synfig::Point pos, float radius)
 {
-	GuideList::iterator iter,best(guide_list_y_.end());
+	has_guide_x = false;
+	has_guide_y = false;
+	GuideList::iterator iter,best(guide_list_.end());
+	second_best_guide_match = guide_list_.end();
```

Comment:
I still don't understand why `has_guide_x`, `has_guide_y` and `second_best_guide_match` are needed.
Even if they are, they shouldn't be property members of `Duckmatic`.
You are making it stateful:
If somewhere `find_guide()` is called, and another thread call it too, this would screw up these values.

Reply:
`has_guide_x` and `has_guide_y` don't need to be class members I removed them. And for`second_best_guide_match` instead of it being a class member I changed it so that if we need to know it we can pass it to the `find_guide` method.

p.s. long story short removed them from being class members